taskchampion/xtask/src/main.rs

```diff
@@ -3,29 +3,123 @@
+
+// Increment length of array when adding tuples of (PATH, REGEX).
+const MSRV_PATH_REGEX: [(&str, &str); 1] = [(
+    "xtask/Cargo.toml",
+    r#"rust-version = "[0-9]+("|\.|[0-9])+""#,
+)];
 
 pub fn main() -> anyhow::Result<()> {
-    let arg = env::args().nth(1);
-    match arg.as_deref() {
-        Some("codegen") => codegen(),
-        Some(arg) => anyhow::bail!("unknown xtask {}", arg),
-        _ => anyhow::bail!("unknown xtask"),
+    let manifest_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR")?);
+    let workspace_dir = manifest_dir.parent().unwrap();
+    let arguments: Vec<String> = env::args().collect();
+
+    if arguments.len() < 2 {
+        anyhow::bail!("xtask: Valid arguments are: `codegen`, `msrv <version x.y>`");
+    }
+
+    match arguments[1].as_str() {
+        "codegen" => codegen(workspace_dir),
+        "msrv" => msrv(arguments, workspace_dir),
+        _ => anyhow::bail!("xtask: unknown xtask"),
     }
 }
 
 /// `cargo xtask codegen`
 ///
 /// This uses ffizz-header to generate `lib/taskchampion.h`.
-fn codegen() -> anyhow::Result<()> {
-    let manifest_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
-    let workspace_dir = manifest_dir.parent().unwrap();
+fn codegen(workspace_dir: &Path) -> anyhow::Result<()> {
     let lib_crate_dir = workspace_dir.join("lib");
     let mut file = File::create(lib_crate_dir.join("taskchampion.h")).unwrap();
     write!(&mut file, "{}", ::taskchampion_lib::generate_header()).unwrap();
 
     Ok(())
 }
+
+/// `cargo xtask msrv (X.Y)`
+///
+/// This checks and updates the Minimum Supported Rust Version for all files specified in MSRV_PATH_REGEX`.
+/// Each line where the regex matches will have all values of the form `#.##` replaced with the given MSRV.
+fn msrv(args: Vec<String>, workspace_dir: &Path) -> anyhow::Result<()> {
+    // check that (X.Y) argument is (mostly) valid:
+    if !args[2].chars().all(|c| c.is_numeric() || c == '.') {
+        anyhow::bail!("xtask: Invalid argument format. Xtask MSRV argument takes the form \"X.Y(y)\", where XYy are numbers. eg: `cargo run xtask MSRV 1.68`");
+    }
+
+    // set regex for replacing version number only within the pattern found within a line
+    let re_msrv_version = Regex::new(r"([0-9]+(\.|[0-9]+|))+")?;
+
+    // for each file in const paths tuple
+    for msrv_file in MSRV_PATH_REGEX {
+        let mut is_pattern_in_file = false;
+
+        let file_path = workspace_dir.join(msrv_file.0);
+        let path: &Path = Path::new(&file_path);
+        if !&path.exists() {
+            anyhow::bail!("xtask: path does not exist {}", &path.display());
+        };
+
+        let mut file: File = File::options().read(true).write(true).open(path)?;
+        let reader = BufReader::new(&file);
+
+        // set search string and the replacement string for version number content
+        let re_msrv_pattern = Regex::new(msrv_file.1)?;
+        let version_replacement_string = &args[2];
```

Comment:
This doesn't change in each iteration of the loop, so it probably makes sense to move this variable up to the top where you check the content of this argument.